PhpAmqpLib/Connection/AMQPSocketConnection.php

```diff
@@ -29,9 +31,11 @@ public function __construct(
         $login_response = null,
         $locale = 'en_US',
         $timeout = 3,
-        $keepalive = false
+        $read_write_timeout = 3,
+        $keepalive = false,
+        $heartbeat = 0
     ) {
-        $io = new SocketIO($host, $port, $timeout, $keepalive);
+        $io = new SocketIO($host, $port, $timeout, $read_write_timeout, $heartbeat, $keepalive);
```

Comment:
Unfortunately this is a BC break. It should only be added in a new major version.